#! /bin/sh /usr/share/dpatch/dpatch-run
## 30_coredumps.dpatch by Florian Ragwitz <rafl@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Don't dump core on fatal errors (Closes: #419998).

@DPATCH@
diff -urNad trunk~/src/xmms/log.c trunk/src/xmms/log.c
--- trunk~/src/xmms/log.c	2007-05-21 17:12:27.000000000 +0200
+++ trunk/src/xmms/log.c	2007-06-18 15:19:58.149938355 +0200
@@ -20,6 +20,7 @@
  */
 
 #include <stdio.h>
+#include <stdlib.h>
 #include <glib.h>
 #include "xmmspriv/xmms_log.h"
 
@@ -29,7 +30,7 @@
 void
 xmms_log_init (gint verbosity)
 {
-	g_log_set_handler (NULL, G_LOG_LEVEL_MASK, xmms_log_handler,
+	g_log_set_handler (NULL, G_LOG_LEVEL_MASK | G_LOG_FLAG_FATAL, xmms_log_handler,
 	                   GINT_TO_POINTER (verbosity));
 
 	xmms_log_info ("Initialized logging system :)");
@@ -66,4 +67,8 @@
 
 	printf ("%s: %s\n", level, message);
 	fflush (stdout);
+
+	if (log_level & G_LOG_LEVEL_ERROR) {
+		exit (EXIT_FAILURE);
+	}
 }
